[qpid-proton] `find_package` is broken

**Describe the bug**
`find_package(qpid-proton REQUIRED CONFIG)` does not work since there is no file `qpid-protonConfig.cmake`. `find_package(Proton REQUIRED CONFIG)` and `find_package(ProtonCpp REQUIRED CONFIG)` do not work either out of the box. Setting `Proton_DIR` and  `ProtonCpp_DIR` so that their config files are found helps, but then there is another failure because `_IMPORT_PREFIX` is not set in the `Config` files and the `IMPORTED_LOCATION_DEBUG` does not point to the debug libraries. Hence, it would seem that the original config files are not what `vcpkg_cmake_config_fixup` expect them to be.

Another unrelated problem is a warning that `BUILD_PYTHON` and `BUILD_RUBY` are CMake variables that are not used. I mention it cause I'll submit a pull request that will solve the main issue, and I will also take the opportunity to remove this warning.

**Environment**
- OS: Experienced on both Windows and Linux (Ubuntu 20.04)
- CMake version: 3.22.2

**To Reproduce**
Steps to reproduce the behavior:
1. Adds the port `qpid-proton` to `vcpkg.json`
2. Try to add `find_package(qpid-proton REQUIRED CONFIG)`, `find_package(Proton REQUIRED CONFIG)` or `find_package(ProtonCpp REQUIRED CONFIG)` to your CMake file
3. Run `cmake -B debug -G Ninja -S .` and see the warning _and then_ the error.
4. Set  `Proton_DIR` and  `ProtonCpp_DIR` so that the config files are found
5. Run `cmake -B debug -G Ninja -S .` and see the other error.

**Expected behavior**
`find_package(qpid-proton REQUIRED CONFIG)` should work and there should be no warning

**Failure logs** 
```shell
-- Configuring x64-linux-dynamic-rel
CMake Warning at /home/klalumiere/src/coveo/cpp/mylib/debug/vcpkg_installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:485 (message):
  The following variables are not used in CMakeLists.txt:

      BUILD_PYTHON
      BUILD_RUBY

  Please recheck them and remove the unnecessary options from the
  `vcpkg_cmake_configure` call.

  If these options should still be passed for whatever reason, please use the
  `MAYBE_UNUSED_VARIABLES` argument.
Call Stack (most recent call first):
  buildtrees/versioning/versions/qpid-proton/c6ab0bd896fa44681e2c3d4b325915ddacb38a1a/portfile.cmake:13 (vcpkg_cmake_configure)
  scripts/ports.cmake:142 (include)
```

then

```shell
CMake Error at /home/klalumiere/src/vcpkg/scripts/buildsystems/vcpkg.cmake:788 (_find_package):
  Could not find a package configuration file provided by "qpid-proton" with
  any of the following names:

    qpid-protonConfig.cmake
    qpid-proton-config.cmake

  Add the installation prefix of "qpid-proton" to CMAKE_PREFIX_PATH or set
  "qpid-proton_DIR" to a directory containing one of the above files.  If
  "qpid-proton" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  CMakeLists.txt:67 (find_package)
```

or

```shell
CMake Error at /home/klalumiere/src/vcpkg/scripts/buildsystems/vcpkg.cmake:788 (_find_package):
  Could not find a package configuration file provided by "Proton" with any
  of the following names:

    ProtonConfig.cmake
    proton-config.cmake

  Add the installation prefix of "Proton" to CMAKE_PREFIX_PATH or set
  "Proton_DIR" to a directory containing one of the above files.  If "Proton"
  provides a separate development package or SDK, be sure it has been
  installed.
```

and after setting `Proton_DIR`, 

```shell
  Imported target "Proton::core" includes non-existent path

    "/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Error in CMakeLists.txt:
  Imported target "Proton::core" includes non-existent path

    "/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



CMake Warning at /home/klalumiere/src/vcpkg/scripts/buildsystems/vcpkg.cmake:578 (_add_library):
  Cannot generate a safe runtime search path for target mylib because files in
  some directories may conflict with libraries in implicit directories:

    runtime library [libqpid-proton-core.so] in /lib may be hidden by files in:
      /home/klalumiere/src/coveo/cpp/mylib/debug/vcpkg_installed/x64-linux-dynamic/debug/lib
    runtime library [libqpid-proton-cpp.so] in /lib may be hidden by files in:
      /home/klalumiere/src/coveo/cpp/mylib/debug/vcpkg_installed/x64-linux-dynamic/debug/lib

  Some of these libraries may not be found correctly.
```

**Additional context**
As written above, I'll open a pull request to solve the issue soon.

--- a/ports/qpid-proton/portfile.cmake
+++ b/ports/qpid-proton/portfile.cmake
@@ -16,8 +16,6 @@ vcpkg_cmake_configure(
         -DPYTHON_EXECUTABLE=${PYTHON3}
         -DLIB_SUFFIX=
         -DBUILD_GO=no
-        -DBUILD_RUBY=no
-        -DBUILD_PYTHON=no
         -DENABLE_JSONCPP=ON
         -DCMAKE_DISABLE_FIND_PACKAGE_CyrusSASL=ON
 )
@@ -26,8 +24,24 @@ vcpkg_cmake_install()
 
 vcpkg_copy_pdbs()
 vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake)
+set(configFiles
+    "${CURRENT_PACKAGES_DIR}/share/${PORT}/Proton/ProtonConfig.cmake"
+    "${CURRENT_PACKAGES_DIR}/share/${PORT}/ProtonCpp/ProtonCppConfig.cmake"
+)
+foreach(configFile IN LISTS configFiles)
+    vcpkg_replace_string("${configFile}"
+        "IMPORTED_LOCATION_DEBUG \"\${_IMPORT_PREFIX}/lib"
+        "IMPORTED_LOCATION_DEBUG \"\${_IMPORT_PREFIX}/debug/lib"
+    )
+    vcpkg_replace_string("${configFile}"
+        "debug \${_IMPORT_PREFIX}/lib"
+        "debug \${_IMPORT_PREFIX}/debug/lib"
+    )
+endforeach()
 vcpkg_fixup_pkgconfig()
 
+configure_file(${CMAKE_CURRENT_LIST_DIR}/qpid-protonConfig.cmake
+    ${CURRENT_PACKAGES_DIR}/share/${PORT}/qpid-protonConfig.cmake COPYONLY)
 file(RENAME "${CURRENT_PACKAGES_DIR}/share/proton/LICENSE.txt"
             "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright")
 

--- a/ports/qpid-proton/qpid-protonConfig.cmake
+++ b/ports/qpid-proton/qpid-protonConfig.cmake
@@ -1,0 +1,4 @@
+get_filename_component(PACKAGE_PREFIX_DIR "${CMAKE_CURRENT_LIST_DIR}/../../" ABSOLUTE)
+set(_IMPORT_PREFIX "${PACKAGE_PREFIX_DIR}")
+include(${CMAKE_CURRENT_LIST_DIR}/Proton/ProtonConfig.cmake)
+include(${CMAKE_CURRENT_LIST_DIR}/ProtonCpp/ProtonCppConfig.cmake)

--- a/ports/qpid-proton/vcpkg.json
+++ b/ports/qpid-proton/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "qpid-proton",
   "version": "0.32.0",
-  "port-version": 2,
+  "port-version": 3,
   "description": "Qpid Proton is a high-performance, lightweight messaging library.",
   "homepage": "https://github.com/apache/qpid-proton",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5606,7 +5606,7 @@
     },
     "qpid-proton": {
       "baseline": "0.32.0",
-      "port-version": 2
+      "port-version": 3
     },
     "qscintilla": {
       "baseline": "2.12.0",

--- a/versions/q-/qpid-proton.json
+++ b/versions/q-/qpid-proton.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "109a9bb4fac5d5c5a52cd620d7ddb22da6fe47f7",
+      "version": "0.32.0",
+      "port-version": 3
+    },
+    {
       "git-tree": "c6ab0bd896fa44681e2c3d4b325915ddacb38a1a",
       "version": "0.32.0",
       "port-version": 2

